### ISSUE TITLE
QuerydslDefaultBinding's analysis of collections

### ISSUE THREAD

**Issue by rxxy** [spring-projects/spring-data-commons#2839]
https://github.com/spring-projects/spring-data-commons/blob/2937d620ad952e32a0fc0009becd9e880b0b9003/src/main/java/org/springframework/data/querydsl/binding/QuerydslDefaultBinding.java#L61

The value here is Collection<Collection\<po>>
The element here is Collection\<po>

The constructed predicate is wrong

https://github.com/spring-projects/spring-data-commons/issues/2834

**Comment by mp911de** [comment]
You're right, that Hibernate leniently accepted `Set<Set<…>>` and now it no longer does. We need to fix the issue, using the original ticket you filed at #2834
